### PR TITLE
feat: Log SAUCECTL_INSTALL_BINARY and SAUCECTL_INSTALL_BINARY_MIRROR

### PR DESCRIPTION
**Pull request by tianfeng92**
## Proposed changes

Given `SAUCECTL_INSTALL_BINARY` set. Error msg:

```
Fetching saucectl binary
Installation failed: Error: failed to download: AxiosError: Request failed with status code 404
Please check that you have access to the URL provided by the SAUCECTL_INSTALL_BINARY environment variable: https://example.com/test.
```

Given `SAUCECTL_INSTALL_BINARY_MIRROR` set. Error msg:
```
Fetching saucectl binary
Installation failed: Error: failed to download: AxiosError: Request failed with status code 404
Please check that you have access to the URL provided by the SAUCECTL_INSTALL_BINARY_MIRROR environment variable: https://example.com/test
```

Given not set customized source. Error msg:
```
Fetching saucectl binary
Installation failed: Error: failed to download: Error: getaddrinfo ENOTFOUND github.com
Please check that you have access to https://github.com/saucelabs/saucectl/releases
```

### CODE REVIEW

File: install.js [hunk 1]
```diff
@@ -18,9 +29,25 @@ async function install() {
...
     .catch((e) => {
       console.error(`Installation failed: ${e}`);
-      console.error(
-        `Check that you have access to https://github.com/saucelabs/saucectl/releases or format of the SAUCECTL_INSTALL_BINARY environment variable is correct\n\n`,
-      );
+      const binarySource = process.env.SAUCECTL_INSTALL_BINARY;
+      const binarySourceMirror = process.env.SAUCECTL_INSTALL_BINARY_MIRROR;
+      if (binarySource) {
+        console.error(
+          `Please verify that you have access to the SAUCECTL_INSTALL_BINARY environment variable (${sanitizeURL(
```
Comment (mhan83) [inline install.js]: This is kind of awkwardly phrased. It sounds like the problem is the env variable couldn't be read, but the real problem is the url *defined* by the env variable couldn't be accessed.

Comment (mhan83) [reply install.js]: You're also not distinguishing between a network error and any other kind of error. So these error messages might be printed for a totally unrelated error.

File: install.js [hunk 2]
```diff
@@ -1,5 +1,16 @@
 const { binWrapper } = require('./index.js');
 
+function sanitizeURL(inputURL) {
+  const parsedURL = new URL(inputURL);
```
Comment (mhan83) [inline install.js:4]: If `inputURL` is unparseable and an error is thrown, is the default error message detailed enough that the user will know what to do?

Comment (tianfeng92) [reply install.js:4]: Checking if `inputURL` is indeed required. While thinking about it, I think the best time is before fetching the binary. So I put the check here https://github.com/saucelabs/node-saucectl/pull/28/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R13

**Review by mhan83 — APPROVED** [review]

**Pull request merged**